### PR TITLE
Tag Sundials v0.3.0

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
This version includes @acroy's large PR, a large breaking change. The API in this version is still the same as in the previous version, but the improved API, coupled with a bump in the Sundials version change, is still a ways off. So because this version has large bug fixes but has a compatible API with code from before, we are tagging this for the time being so that we can take our time with the API PRs.

### CODE REVIEW

File: Sundials/versions/0.3.0/requires [hunk 1]
```diff
@@ -0,0 +1,3 @@
+julia 0.4
+BinDeps 0.3
+Compat
```
Comment (tkelman) [inline Sundials/versions/0.3.0/requires]: should declare a minimum version dependency that implements the features you need

**Comment by ararslan** [comment]
Looks like you need to push the tag to the repository. See https://travis-ci.org/JuliaLang/METADATA.jl/builds/168380522#L209

**Comment by ChrisRackauckas** [comment]
No it's there: https://github.com/JuliaDiffEq/Sundials.jl/releases

It took me awhile to find out that I had to do it manually though.

**Comment by ararslan** [comment]
Oh, I guess the Travis build just needs to be restarted then.

If you use PkgDev, the tags are pushed upstream for you. Highly recommended 👍

**Comment by ChrisRackauckas** [comment]
> If you use PkgDev, the tags are pushed upstream for you. Highly recommended 👍

I used PkgDev on Windows, so the tags aren't pushed upstream for you... highly not recommended 👎

**Comment by tkelman** [comment]
Report an issue.

**Comment by ChrisRackauckas** [comment]
Yeah I always forget to. Sensible minimum version requirements were added. I'll go report the issue right now.

File: Sundials/versions/0.3.0/requires [hunk 2]
```diff
@@ -0,0 +1,3 @@
+julia 0.4
+BinDeps 0.4.3
```
Comment (tkelman) [inline Sundials/versions/0.3.0/requires]: which is it, 0.4.3 or 0.4.5 https://github.com/JuliaDiffEq/Sundials.jl/commit/0e13eddd9095525cfc87185170ee346d7e233678? those are both very new, is there actually a brand new feature or bugfix in BinDeps you're relying on?

Comment (tkelman) [inline Sundials/versions/0.3.0/requires]: Ok. Not much has changed in BinDeps for some time, it's primarily been deprecation fixes.

**Pull request merged**